### PR TITLE
Add the client cert and key support to HttpTransport

### PR DESCRIPTION
**Pull request by grammy-jiang**
Enable 2-way SSL authentication support in sentry-sdk #3256 

The document of [self-hosted Sentry](https://develop.sentry.dev/self-hosted/#productionalizing) strongly recommends using a load balancer in front of Sentry. And when the load balancer enables two-way SSL authentication, this pull request makes sentry-sdk allow to pass the client cert and key file to [urllib3.PoolManager](https://urllib3.readthedocs.io/en/stable/advanced-usage.html#client-certificates).

This pull request adds two new key arguments to [`sentry_sdk.init`](https://github.com/getsentry/sentry-python/blob/2.8.0/sentry_sdk/hub.py#L96-L106):

```python
import sentry_sdk

sentry_sdk.init(
    dsn="https://examplePublicKey@o0.ingest.sentry.io/0",
    ...,
    ca_certs="THE PATH OF ROOT CERTIFICATE",
    # The two new key arguments
    cert_file="THE PATH OF CERTIFICATE",
    key_file="THE PATH OF KEY",
)
```

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

**Review by szokeasaurusrex — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Before we can merge this, you also need to make sure to add the new options [here](https://github.com/getsentry/sentry-python/blob/ee84c81bd00ee9286cdc53f4c1980009e0297eb5/sentry_sdk/consts.py#L453), so they show up in the type hint for `sentry_sdk.init`. 

Also, please add tests for your change

File: sentry_sdk/consts.py
Comment (sl0thentr0py) [inline sentry_sdk/consts.py:507]: whenever you add options here, please append to the end of the list since these are also positional.
No one should ideally use them positionally that way but we have to be safe.

**Review by sl0thentr0py — CHANGES_REQUESTED** [review]

File: sentry_sdk/consts.py
Comment (grammy-jiang) [reply sentry_sdk/consts.py]: Thanks for your comment. `cert_file` and `key_file` are moved to the end of the arguments Now.

**Review by sl0thentr0py — APPROVED** [review]
ty!

**Comment by sl0thentr0py** [comment]
@szokeasaurusrex needs to approve this too because his requests changes status is blocking the merge

**Review by szokeasaurusrex — APPROVED** [review]

**Pull request merged**
